File: 4119-MinimumDistanceBetweenThreeEqualElementsIi/4119-MinimumDistanceBetweenThreeEqualElementsIi.cpp
// Last updated: 20/07/2026, 10:40:04
class Solution {
public:
    int minimumDistance(vector<int>& nums) {
        unordered_map<int,vector<int>> mp;
        int mini=INT_MAX;
        for(int i=0;i<nums.size();i++){
            int num=nums[i];
            if(mp.find(num)==mp.end()){
                mp[num]=vector<int>();
            }
            mp[num].push_back(i);
            if(mp[num].size()>3){
                mp[num].erase(mp[num].begin());
            }   
            if(mp[num].size()==3){
                mini=min(mini,2*(mp[num][2]-mp[num][0]));
            }
        }
        if(mini==INT_MAX){
            return -1;
        }
        return mini;
    }
};

File: 4119-MinimumDistanceBetweenThreeEqualElementsIi/4119-MinimumDistanceBetweenThreeEqualElementsIi.py
# Last updated: 20/07/2026, 10:40:11
class Solution:
    def minimumDistance(self, nums: List[int]) -> int:
        mp={}
        mini=float('inf')
        for i in range(len(nums)):
            num=nums[i]
            if num not in mp:
                mp[num]=[]
            mp[num].append(i)
            if len(mp[num])>3:
                mp[num].pop(0)
            if len(mp[num])==3:
                mini=min(mini,2*(mp[num][2]-mp[num][0]))
        if mini==float('inf'):
            return -1
        return mini